Remove commented-out function definitions and delete files with no executable code

rileyjmurray opened the pull request:
This PR aims to resolve #428.

At time of opening this PR, the changes might be over-zealous, so I'm happy to roll some changes back. That said, I suggest we be aggressive in removing this kind of unused text. Here's a proposal for how to decide what to actually keep and what to remove:
* We can/should keep simple reference implementations of functions that are important but _very_ complicated.
    * Example: there's an implementation of Levenberg-Marquardt from Wikipedia that I've kept. I think this is valuable because our actual LM code is very complicated, so having a reference is helpful for maintainability.
* We could keep text that's related to memory profiling.
    * Rationale: The question of whether or not to bother trying to profile memory usage in a certain location requires a fair amount of expertise, so I think keeping this kind of text could serve as a flag for suspected hot code paths.
* Commented-out pyx code should be removed unless we have a very good reason for keeping it.

Another source of criteria for figuring out what to keep is to look for a short note at the beginning of some commented-out code block. Often there are notes that say "FUTURE," "Removed," "unused," "REMOVE," "Debugging," etc...

Other, miscellaneous changes
* Removed some unused imports.

rileyjmurray commented:
Hi @pcwysoc, @kmrudin, @kevincyoung, @adhumu, and @sserita. This pull request is a big spring-cleaning for commented-out code. You're all [designated as code-owners](https://github.com/sandialabs/pyGSTi/blob/master/.github/CODEOWNERS) for files that are affected by this pull request. Please review the changes for files you "own" and let me know if you object to removing specific blocks of commented-out code.

Things to keep in mind:
* TODOs can be moved to GitHub issues. You can paste code in the GitHub issue as a starting point, for later reference.
* You can replace massive comment-out code blocks with a short comment, like the following
   ```
   # NOTE: The FancyClass is admittedly very complicated. We used to have an equivalent SimpleClass, but
   # for maintainability reasons we removed it. The old SimpleClass can be found at
   # https://github.com/sandialabs/pyGSTi/blob/d27e4e688a64914a0b7aaec1de91aa56f7ce70c2/pygsti/drivers/longsequence.py#L153
   ```
    -- I've used a dummy link for reference.
* If you really want to keep a commented-out code block, please at least add your name and date so it's obvious who made the call to keep it and when. (There's semi-automated tools for this, but they aren't as reliable as old-fashioned comments.)
* Use triple-ticks or triple-quotes for gigantic comments, rather than hundreds of lines that start with the single-line comment character ``#``. Using the former styles makes it possible to collapse irrelevant text in most code editors.

pcwysoc reviewed (approved):
I have no objection to the removed comments on the files for which I'm a code owner. Someone may wish to put the code for the old RB plot somewhere however if it still has utility. I defer to @jordanh6 or @tjproct on this.

adhumu reviewed (approved):
Ditto what Piper said. Stefan and I were working on changing parts of ForwardSims in a different branch and on my local computer. This part of the code is still in development. I'm happy to approve these changes, thanks Riley!

coreyostrove commented on pygsti/modelmembers/povms/denseeffect.py:
GitHub diff is unclear, but this file is getting deleted, right?

rileyjmurray replied:
yup! Getting deleted.

coreyostrove commented on pygsti/models/model.py:
Might need to merge in latest develop. This code was merged in with #482 so shouldn’t be showing up in the diff.

rileyjmurray replied:
Weird. I just tried to rebase from develop, and git says that this branch is already up to date. I'll poke around.

rileyjmurray replied:
I think the GitHub summary of changes might be messed up. I experimented by creating a copy of the current develop branch and then merging in this branch to that copy. The summary of changes in the command line interface didn't include additions to this file.

coreyostrove commented on pygsti/tools/fogitools.py:
```diff
@@ -732,41 +732,6 @@ def resolve_norm_order(vecs_to_normalize, label_lists, given_norm_order):
     return (fogi_dirs, fogi_meta, dep_fogi_dirs, dep_fogi_meta)
 
 
-#def create_fogi_dir_labels(fogi_opsets, fogi_dirs, fogi_rs, fogi_gaugespace_dirs, errorgen_coefficients):
```
I want to keep this. The FOGI code is still under active development and it isn’t immediately obvious we won’t need to revive this in the near-enough future.

rileyjmurray replied:
Roger that! I've reverted changes to this file.

coreyostrove commented on pygsti/tools/fogitools.py:
```diff
@@ -865,190 +830,6 @@ def _create_errgen_op(vec, list_of_mxs):
     return ret
 
 
-#An alternative but inferior algorithm for constructing FOGI quantities: Keep around for checking/reference or REMOVE?
```
Same as above, still under active development so I am not ready to jettison this as flotsam yet.

coreyostrove commented on test/unit/objects/test_prefixtable.py:
I will likely be adding real tests for this module as part of a pending PR I have. If this is a full deletion let’s keep this file for now to avoid merge issues.

rileyjmurray replied:
Change reverted!

coreyostrove commented:
Whoops, forgot to add the top-level comment for my review.

TLDR: Good work @rileyjmurray. It will be a while till most of us can vie for the crown of most lines written, but at this pace you’re definitely in the running for most lines of pyGSTi deleted! I’ll defer judgement to other code owners for stuff they manage (i.e. a lack of comment may just mean I am waiting for one of them to decide whether to chime in rather than explicit approval), but have added a few comments regarding code I would like to keep, as well as a few minor suggestion for additional deletion related clean-up.

rileyjmurray commented on pygsti/algorithms/germselection.py:
Done!

rileyjmurray commented on pygsti/evotypes/densitymx_slow/effectreps.py:
```diff
@@ -105,11 +105,6 @@ def __init__(self, povm_factors, effect_labels, state_space):
         self.factor_effects_have_changed()
 
-    #TODO: fix this:
-    #def __reduce__(self):
```
Not sure. I'll investigate.

Possibly related: I ran into issues with unimplemented pickling functions in my work on faster pytorch-based forward simulators. See the specific commit here: https://github.com/sandialabs/pyGSTi/pull/479/commits/62f37ec58e7845f93fd7bda72fb8eef6692ad1c0.

rileyjmurray commented on pygsti/extras/interpygate/process_tomography.py:
Not sure why I made this change. Reverting.

sserita commented on pygsti/algorithms/mirroring.py:
```diff
@@ -21,92 +21,6 @@
 
 from . import randomcircuit as _rc
 
-# ### TODO: THIS IS TIMS OLD CODE WHICH SHOULD PERHAPS ALSO BE AN OPTION IN THE `CREATE_MIRROR_CIRCUIT` FUNCTION
```
Tim's old mirroring code.

sserita commented on pygsti/algorithms/randomcircuit.py:
```diff
@@ -198,139 +198,6 @@ def find_all_sets_of_compatible_two_q_gates(edgelist, n, gatename='Gcnot', aslab
     return co2Qgates
 
 
-# TJP: I am not aware this is ever used anymore, and it's functionality is possibly even included in the
```
More Tim RC code

sserita commented on pygsti/algorithms/rbfit.py:
```diff
@@ -17,132 +17,6 @@
 from pygsti.tools import rbtools as _rbt
 
 
-# Obsolute function to be deleted.
```
Tim RB fitting.

sserita commented on pygsti/baseobjs/polynomial.py:
```diff
@@ -558,574 +543,6 @@ def to_rep(self):  # , max_num_vars=None not needed anymore -- given at __init__
 Polynomial = FASTPolynomial
 
 
-# class SLOWPolynomial(dict):  # REMOVE THIS CLASS (just for reference)
```
TODO: turn the slow polynomial into a unit test for FASTPolynomial (to be Polynomial)

rileyjmurray replied:
Done. (Mostly. I didn't actually write tests. But I moved this class into the appropriate test file.)

sserita commented on pygsti/evotypes/chp/statereps.py:
```diff
@@ -87,25 +77,3 @@ def reps_have_changed(self):
-#            + self.op_rep.chp_ops(seed_or_state=seed_or_state)
-
-# TODO: Untested, only support computational and composed for now
-#class StateRepTensorProduct(StateRep):
```
For Stefan to check out

rileyjmurray replied:
@sserita did you look into this? If not, could you open a GitHub issue articulating the question that needs investigation?

sserita replied:
It is fine to remove this without further action. I have a personal vested interest in the CHP code so wanted to look into it a bit further, but there's no specific actionable issues to articulate based on the code you are removing here.

sserita commented on pygsti/evotypes/densitymx/opreps.pyx:
```diff
@@ -727,24 +727,6 @@ cdef class OpRepExpErrorgen(OpRep):
 
 
-#TODO: can add this after creating OpCRep_IdentityPlusErrorgen if it seems useful
-#cdef class OpRepIdentityPlusErrorgen(OpRep):
```
+1 to look into this

rileyjmurray replied:
@sserita @coreyostrove did you do the looking? If not, maybe we can open a GitHub issue to keep track of this TODO?

rileyjmurray replied:
Ping @coreyostrove and @sserita. Do we want to open an issue to track this TODO, or have you resolved it to your satisfaction?

coreyostrove replied:
Just looked, this should be resuscitated. If I understand the code correctly then this will result in an error when using this modelmember if the representation type is '1+L'. There is an implementation of this representation in `densitymx_slow`, so an error would get hit when using the cython module (though that is the default when present).

rileyjmurray replied:
Gotcha. I'll revert this change and open a new issue.

sserita commented on pygsti/extras/interpygate/core.py:
```diff
@@ -241,15 +241,6 @@ def create_object(self, args=None, sslbls=None):
         return InterpolatedDenseOp(target_op, self.base_interpolator, self.aux_interpolator, self.to_vector(),
                                    _np.array(args), self._argument_indices)
 
-    #def write(self, dirname):
```
Check with Kevin and Brandon, might still be useful.

sserita commented on pygsti/forwardsims/termforwardsim.py:
```diff
@@ -416,105 +406,6 @@ def _bulk_fill_hprobs_atom(self, array_to_fill, dest_param_slice1, dest_param_sl
                 hpolys[0], hpolys[1], self.model.to_vector(), (nEls, len(wrtInds1), len(wrtInds2)))
         _fas(array_to_fill, [slice(0, array_to_fill.shape[0]), dest_param_slice1, dest_param_slice2], hprobs)
 
-    #DIRECT FNS - keep these around, but they need to be updated (as do routines in fastreplib.pyx)
```
Erik's direct compute reference for TermForwardSimulator

sserita replied:
Maybe a candidate to be update + test

rileyjmurray replied:
@sserita, can you clarify what you mean by "update + test"?

sserita commented on pygsti/forwardsims/termforwardsim_calc_statevec.pyx:
```diff
@@ -1531,654 +1531,3 @@ def circuit_achieved_and_max_sopm(fwdsim, rholabel, elabels, circuit, repcache,
 
     return achieved_sopm, max_sopm
-
-
```
See note in pygsti/forwardsims/termforwardsim.py

rileyjmurray replied:
@sserita, do you mean the note which I've commented on here: https://github.com/sandialabs/pyGSTi/pull/452/files/00982b291c80a972790cf09dfaa4cf556ad1d9da#r1878653983 ?

sserita commented on pygsti/io/mongodb.py:
```diff
@@ -158,16 +158,10 @@ def read_auxtree_from_mongodb_doc(mongodb, doc, auxfile_types_member='auxfile_ty
     cur_typ = subtypes[0]
     next_typ = ':'.join(subtypes[1:])
 
-    # In FUTURE maybe we can implement "quick loading" from a MongoDB, but currently `quick_load` does nothing
```
Leave this comment

rileyjmurray replied:
Done

sserita commented on pygsti/modelmembers/operations/embeddedop.py:
```diff
@@ -276,17 +276,6 @@ def to_dense(self, on_space='minimal'):
         numpy.ndarray
         """
 
-        #FUTURE: maybe here or in a new "tosymplectic" method, could
```
Leave this

rileyjmurray replied:
Done

sserita commented on pygsti/modelmembers/operations/lindbladerrorgen.py:
```diff
@@ -1223,56 +1194,6 @@ def transform_inplace(self, s):
             raise ValueError("Invalid transform for this LindbladErrorgen: type %s"
                              % str(type(s)))
 
-    #I don't think this is ever needed
```
Note: Lindblad Errorgen SPAM transform code

sserita commented on pygsti/modelmembers/povms/denseeffect.py:
```diff
@@ -1,142 +0,0 @@
-
-
-#UNUSED - I think we can remove this
-#class DensePOVMEffect(_POVMEffect):
```
Note: DensePOVMEffect reference

sserita commented on pygsti/modelmembers/povms/marginalizedpovm.py:
```diff
@@ -207,85 +207,6 @@ def __reduce__(self):
                                    self.sslbls_after_marginalizing),
                 {'_gpindices': self._gpindices})  # preserve gpindices (but not parent)
 
-    #May need to implement this in future if we allow non-static MarginalizedPOVMs
```
Note: non-static MarginalizedPOVM ideas

sserita commented on pygsti/modelmembers/term.py:
```diff
@@ -272,16 +272,6 @@ def __mul__(self, x):
     def __rmul__(self, x):
         return self.__mul__(x)
 
-    #Not needed - but we would use this if we changed
```
Leave this

rileyjmurray replied:
Done

sserita commented on pygsti/modelpacks/stdtarget.py:
```diff
@@ -42,127 +42,6 @@ def _get_cachefile_names(std_module, param_type, simulator, py_version):
     raise ValueError("No cache files used for param-type=%s" % param_type)
 
 
-# XXX is this used?
```
This was used for modelpack generation. Fine to remove but note TermForwardSim caching for model packs

rileyjmurray replied:
To clarify, @sserita, would you like the source code to include a comment of some kind given that we'll be removing this?

sserita replied:
No, the note is a "meta" one. When we merge this in, I'll go through and collect all the things we removed here and make a note of them (maybe even in the release notes) so we can come back to reference it.

sserita commented on pygsti/models/memberdict.py:
```diff
@@ -45,18 +45,6 @@ def __setitem__(self, key, val):
-    #    items = [(k,v) for k,v in self.iteritems()]
-    #    return (_PrefixOrderedDict, (self._prefix, items), None)
-
-    """
```
Remove reduce, but leave docstring

rileyjmurray replied:
@sserita are you sure? The docstring doesn't seem to apply to this class. It applies somewhat to the class below, but that class has more detailed docs.

sserita reviewed (commented):
2/3 of the way check-in

tjproct commented:
I’m guessing I’m meant to be doing something? My main reaction to this pull request is/was “argh I’m pretty sure some of that commented out code should be revived, but I have no time to even think about this”. That’s probably very unhelpful, but that’s how it is right now. -T

From: Stefan Seritan ***@***.***>
Date: Tuesday, November 12, 2024 at 11:53 AM
To: sandialabs/pyGSTi ***@***.***>
Cc: Proctor, Timothy James ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [sandialabs/pyGSTi] Remove commented-out function definitions and delete files with no executable code (PR #452)

@sserita commented on this pull request.

2/3 of the way check-in

________________________________

In pygsti/algorithms/mirroring.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838483466>:

> @@ -21,92 +21,6 @@



 from . import randomcircuit as _rc



-# ### TODO: THIS IS TIMS OLD CODE WHICH SHOULD PERHAPS ALSO BE AN OPTION IN THE `CREATE_MIRROR_CIRCUIT` FUNCTION

Tim's old mirroring code.

________________________________

In pygsti/algorithms/randomcircuit.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838486250>:

> @@ -198,139 +198,6 @@ def find_all_sets_of_compatible_two_q_gates(edgelist, n, gatename='Gcnot', aslab

     return co2Qgates





-# TJP: I am not aware this is ever used anymore, and it's functionality is possibly even included in the

More Tim RC code

________________________________

In pygsti/algorithms/rbfit.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838486967>:

> @@ -17,132 +17,6 @@

 from pygsti.tools import rbtools as _rbt





-# Obsolute function to be deleted.

Tim RB fitting.

________________________________

In pygsti/evotypes/chp/statereps.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838494751>:

> @@ -87,25 +77,3 @@ def reps_have_changed(self):

     def actionable_staterep(self):

         state_rep = self.state_rep.actionable_staterep()

         return self.op_rep.acton(state_rep)

-

-#REMOVE

-#    def chp_ops(self, seed_or_state=None):

-#        return self.state_rep.chp_ops(seed_or_state=seed_or_state) \

-#            + self.op_rep.chp_ops(seed_or_state=seed_or_state)

-

-# TODO: Untested, only support computational and composed for now

-#class StateRepTensorProduct(StateRep):

For Stefan to check out

________________________________

In pygsti/extras/interpygate/core.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838506557>:

> @@ -241,15 +241,6 @@ def create_object(self, args=None, sslbls=None):

         return InterpolatedDenseOp(target_op, self.base_interpolator, self.aux_interpolator, self.to_vector(),

                                    _np.array(args), self._argument_indices)



-    #def write(self, dirname):

Check with Kevin and Brandon, might still be useful.

________________________________

In pygsti/forwardsims/termforwardsim_calc_statevec.pyx<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838516855>:

> @@ -1531,654 +1531,3 @@ def circuit_achieved_and_max_sopm(fwdsim, rholabel, elabels, circuit, repcache,

         max_sopm[i] = max_sum_of_pathmags[i]



     return achieved_sopm, max_sopm

-

-

See note in pygsti/forwardsims/termforwardsim.py

________________________________

In pygsti/io/mongodb.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838517936>:

>      subtypes = typ.split(':')

     cur_typ = subtypes[0]

     next_typ = ':'.join(subtypes[1:])



-    # In FUTURE maybe we can implement "quick loading" from a MongoDB, but currently `quick_load` does nothing

Leave this comment

________________________________

In pygsti/modelmembers/operations/embeddedop.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838521826>:

> @@ -276,17 +276,6 @@ def to_dense(self, on_space='minimal'):

         numpy.ndarray

         """



-        #FUTURE: maybe here or in a new "tosymplectic" method, could

Leave this

________________________________

In pygsti/modelmembers/operations/lindbladerrorgen.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838524509>:

> @@ -1223,56 +1194,6 @@ def transform_inplace(self, s):

             raise ValueError("Invalid transform for this LindbladErrorgen: type %s"

                              % str(type(s)))



-    #I don't think this is ever needed

Note: Lindblad Errorgen SPAM transform code

________________________________

In pygsti/modelmembers/povms/denseeffect.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838526169>:

> @@ -1,142 +0,0 @@

-

-

-#UNUSED - I think we can remove this

-#class DensePOVMEffect(_POVMEffect):

Note: DensePOVMEffect reference

________________________________

In pygsti/modelmembers/povms/marginalizedpovm.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838527082>:

> @@ -207,85 +207,6 @@ def __reduce__(self):

                                    self.sslbls_after_marginalizing),

                 {'_gpindices': self._gpindices})  # preserve gpindices (but not parent)



-    #May need to implement this in future if we allow non-static MarginalizedPOVMs

Note: non-static MarginalizedPOVM ideas

________________________________

In pygsti/modelmembers/term.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838529089>:

> @@ -272,16 +272,6 @@ def __mul__(self, x):

     def __rmul__(self, x):

         return self.__mul__(x)



-    #Not needed - but we would use this if we changed

Leave this

________________________________

In pygsti/modelpacks/stdtarget.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838532340>:

> @@ -42,127 +42,6 @@ def _get_cachefile_names(std_module, param_type, simulator, py_version):

     raise ValueError("No cache files used for param-type=%s" % param_type)





-# XXX is this used?

This was used for modelpack generation. Fine to remove but note TermForwardSim caching for model packs

________________________________

In pygsti/models/memberdict.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838534066>:

> @@ -45,18 +45,6 @@ def __setitem__(self, key, val):

                            "beginning with the prefix '%s'" % self._prefix)

         super(_PrefixOrderedDict, self).__setitem__(key, val)



-    #Handled by derived classes

-    #def __reduce__(self):

-    #    items = [(k,v) for k,v in self.iteritems()]

-    #    return (_PrefixOrderedDict, (self._prefix, items), None)

-

-    """

Remove reduce, but leave docstring

________________________________

In pygsti/baseobjs/polynomial.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838491154>:

> @@ -558,574 +543,6 @@ def to_rep(self):  # , max_num_vars=None not needed anymore -- given at __init__

 Polynomial = FASTPolynomial





-# class SLOWPolynomial(dict):  # REMOVE THIS CLASS (just for reference)

TODO: turn the slow polynomial into a unit test for FASTPolynomial (to be Polynomial)

________________________________

In pygsti/evotypes/densitymx/opreps.pyx<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838498747>:

> @@ -727,24 +727,6 @@ cdef class OpRepExpErrorgen(OpRep):

         return _copy.deepcopy(self)  # I think this should work using reduce/setstate framework TODO - test and maybe put in base class?





-#TODO: can add this after creating OpCRep_IdentityPlusErrorgen if it seems useful

-#cdef class OpRepIdentityPlusErrorgen(OpRep):

+1 to look into this

________________________________

In pygsti/forwardsims/termforwardsim.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838514058>:

> @@ -416,105 +406,6 @@ def _bulk_fill_hprobs_atom(self, array_to_fill, dest_param_slice1, dest_param_sl

                 hpolys[0], hpolys[1], self.model.to_vector(), (nEls, len(wrtInds1), len(wrtInds2)))

         _fas(array_to_fill, [slice(0, array_to_fill.shape[0]), dest_param_slice1, dest_param_slice2], hprobs)



-    #DIRECT FNS - keep these around, but they need to be updated (as do routines in fastreplib.pyx)

Erik's direct compute reference for TermForwardSimulator

________________________________

In pygsti/forwardsims/termforwardsim.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838515811>:

> @@ -416,105 +406,6 @@ def _bulk_fill_hprobs_atom(self, array_to_fill, dest_param_slice1, dest_param_sl

                 hpolys[0], hpolys[1], self.model.to_vector(), (nEls, len(wrtInds1), len(wrtInds2)))

         _fas(array_to_fill, [slice(0, array_to_fill.shape[0]), dest_param_slice1, dest_param_slice2], hprobs)



-    #DIRECT FNS - keep these around, but they need to be updated (as do routines in fastreplib.pyx)

Maybe a candidate to be update + test

—
Reply to this email directly, view it on GitHub<https://github.com/sandialabs/pyGSTi/pull/452#pullrequestreview-2430229882>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFQEBIBJDOEXVBODCBD527L2AI5VXAVCNFSM6AAAAABI6RXALOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMZQGIZDSOBYGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

rileyjmurray commented:
Nah, you're good, Tim. No action items for you!
________________________________
From: tjproct ***@***.***>
Sent: Tuesday, November 12, 2024 1:06 PM
To: sandialabs/pyGSTi ***@***.***>
Cc: Murray, Riley John ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [sandialabs/pyGSTi] Remove commented-out function definitions and delete files with no executable code (PR #452)


I’m guessing I’m meant to be doing something? My main reaction to this pull request is/was “argh I’m pretty sure some of that commented out code should be revived, but I have no time to even think about this”. That’s probably very unhelpful, but that’s how it is right now. -T

From: Stefan Seritan ***@***.***>
Date: Tuesday, November 12, 2024 at 11:53 AM
To: sandialabs/pyGSTi ***@***.***>
Cc: Proctor, Timothy James ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [sandialabs/pyGSTi] Remove commented-out function definitions and delete files with no executable code (PR #452)

@sserita commented on this pull request.

2/3 of the way check-in

________________________________

In pygsti/algorithms/mirroring.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838483466>:

> @@ -21,92 +21,6 @@



from . import randomcircuit as _rc



-# ### TODO: THIS IS TIMS OLD CODE WHICH SHOULD PERHAPS ALSO BE AN OPTION IN THE `CREATE_MIRROR_CIRCUIT` FUNCTION

Tim's old mirroring code.

________________________________

In pygsti/algorithms/randomcircuit.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838486250>:

> @@ -198,139 +198,6 @@ def find_all_sets_of_compatible_two_q_gates(edgelist, n, gatename='Gcnot', aslab

return co2Qgates





-# TJP: I am not aware this is ever used anymore, and it's functionality is possibly even included in the

More Tim RC code

________________________________

In pygsti/algorithms/rbfit.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838486967>:

> @@ -17,132 +17,6 @@

from pygsti.tools import rbtools as _rbt





-# Obsolute function to be deleted.

Tim RB fitting.

________________________________

In pygsti/evotypes/chp/statereps.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838494751>:

> @@ -87,25 +77,3 @@ def reps_have_changed(self):

def actionable_staterep(self):

state_rep = self.state_rep.actionable_staterep()

return self.op_rep.acton(state_rep)

-

-#REMOVE

-# def chp_ops(self, seed_or_state=None):

-# return self.state_rep.chp_ops(seed_or_state=seed_or_state) \

-# + self.op_rep.chp_ops(seed_or_state=seed_or_state)

-

-# TODO: Untested, only support computational and composed for now

-#class StateRepTensorProduct(StateRep):

For Stefan to check out

________________________________

In pygsti/extras/interpygate/core.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838506557>:

> @@ -241,15 +241,6 @@ def create_object(self, args=None, sslbls=None):

return InterpolatedDenseOp(target_op, self.base_interpolator, self.aux_interpolator, self.to_vector(),

_np.array(args), self._argument_indices)



- #def write(self, dirname):

Check with Kevin and Brandon, might still be useful.

________________________________

In pygsti/forwardsims/termforwardsim_calc_statevec.pyx<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838516855>:

> @@ -1531,654 +1531,3 @@ def circuit_achieved_and_max_sopm(fwdsim, rholabel, elabels, circuit, repcache,

max_sopm[i] = max_sum_of_pathmags[i]



return achieved_sopm, max_sopm

-

-

See note in pygsti/forwardsims/termforwardsim.py

________________________________

In pygsti/io/mongodb.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838517936>:

> subtypes = typ.split(':')

cur_typ = subtypes[0]

next_typ = ':'.join(subtypes[1:])



- # In FUTURE maybe we can implement "quick loading" from a MongoDB, but currently `quick_load` does nothing

Leave this comment

________________________________

In pygsti/modelmembers/operations/embeddedop.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838521826>:

> @@ -276,17 +276,6 @@ def to_dense(self, on_space='minimal'):

numpy.ndarray

"""



- #FUTURE: maybe here or in a new "tosymplectic" method, could

Leave this

________________________________

In pygsti/modelmembers/operations/lindbladerrorgen.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838524509>:

> @@ -1223,56 +1194,6 @@ def transform_inplace(self, s):

raise ValueError("Invalid transform for this LindbladErrorgen: type %s"

% str(type(s)))



- #I don't think this is ever needed

Note: Lindblad Errorgen SPAM transform code

________________________________

In pygsti/modelmembers/povms/denseeffect.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838526169>:

> @@ -1,142 +0,0 @@

-

-

-#UNUSED - I think we can remove this

-#class DensePOVMEffect(_POVMEffect):

Note: DensePOVMEffect reference

________________________________

In pygsti/modelmembers/povms/marginalizedpovm.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838527082>:

> @@ -207,85 +207,6 @@ def __reduce__(self):

self.sslbls_after_marginalizing),

{'_gpindices': self._gpindices}) # preserve gpindices (but not parent)



- #May need to implement this in future if we allow non-static MarginalizedPOVMs

Note: non-static MarginalizedPOVM ideas

________________________________

In pygsti/modelmembers/term.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838529089>:

> @@ -272,16 +272,6 @@ def __mul__(self, x):

def __rmul__(self, x):

return self.__mul__(x)



- #Not needed - but we would use this if we changed

Leave this

________________________________

In pygsti/modelpacks/stdtarget.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838532340>:

> @@ -42,127 +42,6 @@ def _get_cachefile_names(std_module, param_type, simulator, py_version):

raise ValueError("No cache files used for param-type=%s" % param_type)





-# XXX is this used?

This was used for modelpack generation. Fine to remove but note TermForwardSim caching for model packs

________________________________

In pygsti/models/memberdict.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838534066>:

> @@ -45,18 +45,6 @@ def __setitem__(self, key, val):

"beginning with the prefix '%s'" % self._prefix)

super(_PrefixOrderedDict, self).__setitem__(key, val)



- #Handled by derived classes

- #def __reduce__(self):

- # items = [(k,v) for k,v in self.iteritems()]

- # return (_PrefixOrderedDict, (self._prefix, items), None)

-

- """

Remove reduce, but leave docstring

________________________________

In pygsti/baseobjs/polynomial.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838491154>:

> @@ -558,574 +543,6 @@ def to_rep(self): # , max_num_vars=None not needed anymore -- given at __init__

Polynomial = FASTPolynomial





-# class SLOWPolynomial(dict): # REMOVE THIS CLASS (just for reference)

TODO: turn the slow polynomial into a unit test for FASTPolynomial (to be Polynomial)

________________________________

In pygsti/evotypes/densitymx/opreps.pyx<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838498747>:

> @@ -727,24 +727,6 @@ cdef class OpRepExpErrorgen(OpRep):

return _copy.deepcopy(self) # I think this should work using reduce/setstate framework TODO - test and maybe put in base class?





-#TODO: can add this after creating OpCRep_IdentityPlusErrorgen if it seems useful

-#cdef class OpRepIdentityPlusErrorgen(OpRep):

+1 to look into this

________________________________

In pygsti/forwardsims/termforwardsim.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838514058>:

> @@ -416,105 +406,6 @@ def _bulk_fill_hprobs_atom(self, array_to_fill, dest_param_slice1, dest_param_sl

hpolys[0], hpolys[1], self.model.to_vector(), (nEls, len(wrtInds1), len(wrtInds2)))

_fas(array_to_fill, [slice(0, array_to_fill.shape[0]), dest_param_slice1, dest_param_slice2], hprobs)



- #DIRECT FNS - keep these around, but they need to be updated (as do routines in fastreplib.pyx)

Erik's direct compute reference for TermForwardSimulator

________________________________

In pygsti/forwardsims/termforwardsim.py<https://github.com/sandialabs/pyGSTi/pull/452#discussion_r1838515811>:

> @@ -416,105 +406,6 @@ def _bulk_fill_hprobs_atom(self, array_to_fill, dest_param_slice1, dest_param_sl

hpolys[0], hpolys[1], self.model.to_vector(), (nEls, len(wrtInds1), len(wrtInds2)))

_fas(array_to_fill, [slice(0, array_to_fill.shape[0]), dest_param_slice1, dest_param_slice2], hprobs)



- #DIRECT FNS - keep these around, but they need to be updated (as do routines in fastreplib.pyx)

Maybe a candidate to be update + test

—
Reply to this email directly, view it on GitHub<https://github.com/sandialabs/pyGSTi/pull/452#pullrequestreview-2430229882>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFQEBIBJDOEXVBODCBD527L2AI5VXAVCNFSM6AAAAABI6RXALOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMZQGIZDSOBYGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

—
Reply to this email directly, view it on GitHub<https://github.com/sandialabs/pyGSTi/pull/452#issuecomment-2471230067>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACRLIFE7TH3LQ4K6GLPHGZ32AI7RVAVCNFSM6AAAAABI6RXALOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZRGIZTAMBWG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

sserita commented on pygsti/tools/rbtheory.py:
```diff
@@ -458,255 +458,6 @@ def R_matrix(model, group, group_to_model=None, weights=None):  # noqa N802
 
-### COMMENTED OUT SO THAT THIS FILE DOESN'T NEED "from .. import construction as _cnst".
-### THIS SHOULD BE ADDED BACK IN AT SOME POINT.
-# def exact_rb_asps(model, group, m_max, m_min=0, m_step=1, success_outcomelabel=('0',),
```
More RB tools in case we need to revive them.

sserita commented on test/unit/objects/test_prefixtable.py:
```diff
@@ -1,62 +0,0 @@
-import numpy as np
```
As Corey mentioned, keep this file for now to avoid merge issues with his branch.

rileyjmurray replied:
Done

sserita reviewed (changes requested):
The majority of my comments are just personal notes for things we should highlight/bookmark in the PR in case we want to come back and revive them later.
Some of my comments are things I would like to be left in, and at least one is an item I need to dig further into to decide.

Thanks for your hard work in help clean up the code, it is super appreciated!

rileyjmurray commented on pygsti/baseobjs/errorgenbasis.py:
```diff
@@ -79,11 +79,6 @@ def label_index(self, label, ok_if_missing=False):
             return None
         return self._label_indices[label]
 
-    #@property
```
@coreyostrove please confirm whether you'd like this comment restored, or if it's okay for me to proceed with removing it.

coreyostrove replied:
You can delete this.

rileyjmurray commented on pygsti/evotypes/densitymx_slow/opreps.py:
```diff
@@ -466,15 +455,6 @@ def set_exp_params(self, mu, eta, m_star, s):
     def exp_params(self):
         return (self.mu, self.eta, self.m_star, self.s)
 
-    #def __reduce__(self):
```
Tracking in https://github.com/sandialabs/pyGSTi/pull/452/files#r1814977466

rileyjmurray commented on pygsti/evotypes/stabilizer_slow/effectreps.py:
```diff
@@ -80,9 +61,6 @@ def __init__(self, op_rep, effect_rep, op_id, state_space):
 
         super(EffectRepComposed, self).__init__(state_space)
 
-    #def __reduce__(self):
```
Tracking in https://github.com/sandialabs/pyGSTi/pull/452/files#r1814977466

rileyjmurray commented on pygsti/evotypes/statevec_slow/effectreps.py:
```diff
@@ -180,9 +180,6 @@ def __init__(self, op_rep, effect_rep, op_id, state_space):
 
         super(EffectRepComposed, self).__init__(effect_rep.state_space)
 
-    #def __reduce__(self):
```
Tracking in https://github.com/sandialabs/pyGSTi/pull/452/files#r1814977466

rileyjmurray commented on pygsti/extras/rb/benchmarker.py:
Ping @tjproct and @jordanh6 -- can you comment on Corey's question?

rileyjmurray replied:
Marking as resolved due to lack of response. There's no harm in keeping this in for now.

tjproct replied:
Ah, sorry. No there's nothing in there that we still need -- go ahead and remove the entire extras/rb module.

rileyjmurray replied:
Amazing! Thanks, Tim.

coreyostrove reviewed (approved):
Approved! Thanks for managing this slog!

sserita reviewed (approved):
I looked through all the unresolved conversations that needed my input. None of them have actions required on this PR, so I'm approving. Thanks for your hard work on this Riley.

As we complete/merge this in, I'll probably go through and put together a list of things that we removed that we might want to revisit in the future (either to fill out the feature or to turn it into tests). I'll include that as a comment on the finalized PR, and probably also summarize it in the 0.9.13 release notes.
